homeassistant/components/plugwise/gateway.py

```diff
@@ -42,17 +46,16 @@ async def async_setup_entry_gw(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         connected = await api.connect()
-    except InvalidAuthentication:
-        LOGGER.error("Invalid username or Smile ID")
-        return False
-    except (ClientConnectionError, PlugwiseException) as err:
-        raise ConfigEntryNotReady(
-            f"Error while communicating to device {api.smile_name}"
-        ) from err
-    except asyncio.TimeoutError as err:
+    except ConnectionFailedError as err:
+        raise ConfigEntryNotReady("Failed to connect to the Plugwise Smile") from err
+    except InvalidAuthentication as err:
+        raise HomeAssistantError("Invalid username or Smile ID") from err
```

Comment:
Why do we raise HomeAssistantError?

Reply:
It has been removed in a follow up PR that implemented the ConfigEntryErrors.